### PR TITLE
Remove 502

### PR DESCRIPTION
**Pull request by costajohnt**

### CODE REVIEW

**Comment by costajohnt** [comment]
Its been refactored a bit, but having all three transmuxes start in that same function is not ideal.  What if this first function sends a success message with the jobid, output path, and outputid created in the function.  This message triggers three separate lambdas, each for a different quality transmux.  Each of these can return their own success or error message so we have everything covered.
4

**Comment by ilyakatz** [comment]
what is the problem with having them all start at the same time?

File: src/BitmovinTransmuxingLambda/index.js [hunk 1]
```diff
@@ -12,7 +12,9 @@ const getRepresentationId = (quality, videoStreamConfigs) => {
   if (config) return config.representationId;
 }
 
-const createOutput = (bitcodin, outputPath, jobId) => {
+const createOutput = (message, bitcodin) => {
+  const outputPath = message.payload.outputUrl.match(/[0-9]{2,}_[A-z 0-9]+$/)[0];
```
Comment (ilyakatz) [inline src/BitmovinTransmuxingLambda/index.js]: is this the same regex as below? probably good idea to put it into a constant

File: src/BitmovinTransmuxingLambda/index.js [hunk 2]
```diff
@@ -23,69 +25,44 @@ const createOutput = (bitcodin, outputPath, jobId) => {
...
+  if(audioStreamConfigs.length > 0) {
+    for (var i = 0, len = audioStreamConfigs.length; i < len; i++) {
+      audioRepresentationIds.push(audioStreamConfigs[i].representationId);
     }
-  );    
+    transmuxConfiguration.audioRepresentationIds = audioRepresentationIds;
+  }
+
+  if(videoStreamConfigs.length > 0) {
+    transmuxConfiguration.videoRepresentationId = getRepresentationId(transmuxConfiguration.filename, videoStreamConfigs);
```
Comment (ilyakatz) [inline src/BitmovinTransmuxingLambda/index.js]: Woof: line too long

File: src/BitmovinTransmuxingLambda/index.js [hunk 3]
```diff
@@ -95,13 +72,22 @@ const createMediumTransmux = createTransmux("medium");
 const createHighTransmux = createTransmux("high");
 
 class BitmovinTransmuxingLambda {
-  createTransmuxes = (message) => {
-    message = JSON.parse(message)
-    const jobId = message.payload.jobId;
-    const outputPath = message.payload.outputUrl.match(/[0-9]{2,}_[A-z 0-9]+$/)[0];
+  start = (message) => {
     const bitcodin = require('bitcodin')(process.env.BITMOVIN_API_TOKEN);
+    const outputPath = message.payload.outputUrl.match(/[0-9]{2,}_[A-z 0-9]+$/)[0];
```
Comment (ilyakatz) [inline src/BitmovinTransmuxingLambda/index.js]: another reuse of regex

File: src/BitmovinTransmuxingLambda/index.js [hunk 4]
```diff
@@ -23,69 +25,44 @@ const createOutput = (bitcodin, outputPath, jobId) => {
...
+  const jobId = message.payload.jobId;
+  const outputPath = message.payload.outputUrl.match(/[0-9]{2,}_[A-z 0-9]+$/)[0];
+
+  return bitcodin.output.list(0, 'finished')
 }
 
-const createTransmux = quality => (bitcodin, outputPath, jobId, outputId) => {
-  let transmuxings = [];
-  let transmuxDetails;
+const getJobDeets = (bitcodin, jobId) => {
```
Comment (ilyakatz) [inline src/BitmovinTransmuxingLambda/index.js]: mind renaming?

File: src/handler.js [hunk 5]
```diff
@@ -5,24 +5,70 @@ require('dotenv').config();
...
+  );
 }
 
 function test() {
   const bit = new BitmovinTransmuxingLambda();
   console.log('Starting new bitmovin transmuxing job');
   const event = jf.readFileSync('event.json');
   const message = processMessage(event);
-  bit.createTransmuxes(message);
+  bit.start(message)
```
Comment (ilyakatz) [inline src/handler.js]: this seems strange, test() and entry() should not repeat so much code, those error conditions should be in a shared method that both test and entry call

File: src/BitmovinTransmuxingLambda/index.js (hunk 3 above)
Comment (costajohnt) [inline src/BitmovinTransmuxingLambda/index.js]: removed duplicate

File: src/handler.js (hunk 5 above)
Comment (costajohnt) [inline src/handler.js]: resolved

Comment (ilyakatz) [inline src/handler.js]: i can live with this for now, in the long terms would be great to figure out how the test method can also do a callback :)

File: src/BitmovinTransmuxingLambda/index.js (hunk 3 above)
Comment (ilyakatz) [inline src/BitmovinTransmuxingLambda/index.js]: i still see it in two places

File: src/handler.js (hunk 5 above)
Comment (costajohnt) [inline src/handler.js]: that would be cool.  not sure if its possible yet

File: src/BitmovinTransmuxingLambda/index.js (hunk 3 above)
Comment (costajohnt) [inline src/BitmovinTransmuxingLambda/index.js]: removed

**Pull request merged**